src/app/core/services/auth/auth.service.ts

```diff
@@ -0,0 +1,38 @@
+import { IUser } from './../../models/user.model';
+import { Injectable } from '@angular/core';
+
+const MOCK_USER: IUser = {
+  firstName: 'Walton',
+  lastName: 'Decker',
+  id: 123
+};
+
+@Injectable({
+  providedIn: 'root'
+})
+export class AuthService {
+  public isAuthenticated: boolean;
+  private user: IUser;
+
+  constructor() {
+    this.isAuthenticated = Boolean(this.user);
```

Comment:
better way to initialize data in OnInit LCH, because it can be absent when constructor calls. not this case, just reminder